modules/reports/models/scheduled_reports.php

```diff
@@ -337,25 +337,25 @@ static function get_period_schedules($period_str)
 	{
 		$schedules = array();
 		$send_date = array();
+		$default_timezone = date_default_timezone_get();
 
 		$db = Database::instance();
 
 		$sql = <<<'SQL'
 		SELECT sr.*, rp.periodname, opt.value AS timezone
 		FROM scheduled_reports sr
 		INNER JOIN scheduled_report_periods rp ON rp.id = sr.period_id
-		INNER JOIN saved_reports_options opt ON opt.report_id = sr.report_id
+		LEFT JOIN saved_reports_options opt ON opt.report_id = sr.report_id
```

Comment:
My SQL skills are kinda rusty, but I do not really understand why this change is here.

Before I assume we got the timezone option from `saved_reports_options` if it existed, and I guess that has not changed with a `LEFT JOIN` or?

Reply:
With an `INNER JOIN`, there has to be a match for the `ON` clause for the row to be returned (i.e. `ON opt.report_id = sr.report_id AND opt.name = 'report_timezone'`. With `LEFT JOIN`, if the ON clause doesn't match for the right table ("saved_report_options"), it will still return the row, but filled with NULL for the missing values from right table.
So effectively, if there was no `report_timezone` option for some reason, with an INNER JOIN, the schedule wouldn't get returned at all.

Reply:
Ahh, ok. Makes sense